src/Controller/Component/SchemaComponent.php

```diff
@@ -364,6 +364,44 @@ public function descendants(string $type): array
         return (array)Hash::get($features, sprintf('descendants.%s', $type));
     }
 
+    /**
+     * Retrieve list of custom properties for a given object type.
+     *
+     * @param string $type Object type name.
+     * @return array
+     */
+    public function customProps(string $type): array
+    {
+        return (array)Cache::remember(
+            CacheTools::cacheKey(sprintf('custom_props_%s', $type)),
+            function () use ($type) {
+                return $this->fetchCustomProps($type);
+            },
+            self::CACHE_CONFIG
+        );
+    }
+
+    /**
+     * Fetch custom properties for a given object type.
+     *
+     * @param string $type Object type name.
+     * @return array
+     */
+    protected function fetchCustomProps(string $type): array
+    {
+        if ($this->getConfig('internalSchema')) {
+            return []; // internal resources don't have custom properties
+        }
+        $query = [
+            'fields' => 'name',
+            'filter' => ['object_type' => $type, 'type' => 'dynamic'],
+            'page_size' => 100,
```

Comment:
100 custom properties is probably enough. However, to be completely safe, this part should iterate over pagination results.

Reply:
Done